### PR TITLE
Add tests for JLanguageMultilang

### PR DESCRIPTION
**Pull request by wilsonge**
There is a small change to allow the application object to be injected in to allow unit tests this can be tested easily by just navigating around Joomla.

Finally as per the doc block I've typecast the result to be a boolean as it was returning the string 0 on false and 1 on true for the enabled state of the plugin (as these are the values stored in the database). Currently you get true and false in the frontend and 0 and 1 in the backend.
## Testing

All the routing in the frontend depends on this class so navigate around and ensure that with plugin enabled and disabled everything works as expected.

In the backend ensure that the modal views when you are selecting a single item in the menu show everything properly

### CODE REVIEW

File: libraries/cms/language/multilang.php [hunk 1]
```diff
@@ -32,8 +34,11 @@ public static function isEnabled()
 		// Status of language filter plugin.
 		static $enabled = false;
 
-		// Get application object.
-		$app = JFactory::getApplication();
+		// Get application object if not injected.
+		if (!$app)
+		{
+			$app = JFactory::getApplication();
```
Comment (Hackwar) [inline libraries/cms/language/multilang.php]: This one will not be tested if you always hand over a $app in the tests. Thus not 100% coverage. :wink:

File: libraries/cms/language/multilang.php [hunk 2]
```diff
@@ -20,11 +20,13 @@ class JLanguageMultilang
...
+	 *
 	 * @return  boolean  True if site is supporting multiple languages; false otherwise.
 	 *
 	 * @since   2.5.4
 	 */
-	public static function isEnabled()
+	public static function isEnabled(JApplicationCms $app = null)
 	{
 		// Flag to avoid doing multiple database queries.
 		static $tested = false;
```
Comment (Hackwar) [inline libraries/cms/language/multilang.php]: I hate this additional variable... I know it makes testing easier, but it feels nasty.

**Comment by Hackwar** [comment]
Looks good. This test brings up 2 issues for me: JLanguageMultilang couples a multi-lang implementation to our language filter plugin and it shows that our language implementation is pretty hacked together... This class is pretty much just bloat. A proper implementation would read that from some general storage and a proper languagefilter implementation would set that flag regardless of the app that is currently running.

**Comment by wilsonge** [comment]
Replaced by https://github.com/joomla/joomla-cms/pull/6479 which removes the application injection

**Pull request closed without merging**